src/main/java/seedu/address/logic/commands/ListOrderCommand.java

```diff
@@ -0,0 +1,92 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.order.Order;
+import seedu.address.model.person.Person;
+
+/**
+ * Lists all orders of the person identified using it's displayed index from the address book.
+ */
+public class ListOrderCommand extends Command {
+
+    public static final String COMMAND_WORD = "listorder";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Lists all orders of the person identified by the index number used in the displayed person list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    private final Index targetIndex;
+
+    /**
+     * Creates a ListOrderCommand to list the orders of the person at the specified {@code Index}.
+     *
+     * @param targetIndex Index of the person in the filtered person list.
+     */
+    public ListOrderCommand(Index targetIndex) {
+        requireNonNull(targetIndex);
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireAllNonNull(model);
+
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personTargeted = lastShownList.get(targetIndex.getZeroBased());
+        List<Order> orderList = personTargeted.getOrders();
+
+        return new CommandResult(createOrderListString(orderList));
+    }
+
+    /**
+     * Creates a string that lists all orders.
+     *
+     * @param orders List of orders to be listed.
+     * @return A string representation of the order list.
+     */
+    private String createOrderListString(List<Order> orders) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("Orders for the selected person:\n");
+
+        // Sort orders by date using a stream
+        List<Order> sortedOrders = orders.stream()
+                .sorted(Comparator.comparing(Order::getDate))
+                .collect(Collectors.toList());
```

Comment:
Currently it looks like you are sorting by date in ascending order. I think sorting it by descending order would be better? So the latest orders are at the top.